src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -2538,6 +2538,8 @@ void sofia_event_callback(nua_event_t event,
 						switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "detaching session %s\n", sofia_private->uuid);
 						if (switch_true(telnyx_force_cdr)) {
 							switch_channel_clear_flag(channel, CF_NO_CDR);
+						} else if (switch_false(telnyx_force_cdr)) {
```

Comment:
Just question.

Who is setting the `telnyx_force_cdr` channel variable? dialplan? 
Is the dialplan doesn't set the `telnyx_force_cdr` channel variable when it get's 401/407?

Reply:
And how about `else`? not `else if`.

Reply:
Dialplan is setting this variable. I am enforcing that they have to explicitly set the telnyx_force_cdr value before we decide if we will disable cdr or not. If they don't set the variable we don't do nothing and use the default behavior

Reply:
Thanks, then this needs to be as `else if`. Clear now. :)